_inc/lib/widgets.php

```diff
@@ -676,4 +676,86 @@ public static function get_registered_widget_object( $id_base ) {
 	public static function validate_id_base( $id_base ) {
 		return ( false !== self::get_registered_widget_object( $id_base ) );
 	}
+
+	/**
+	 * Insert a new widget in a given sidebar.
+	 *
+	 * @param string $widget_id ID of the widget.
+	 *
+	 * @param array $widget_options Content of the widget.
+ 	 *
+ 	 * @param string $sidebar ID of the sidebar to which the widget will be added.
+ 	 *
+ 	 * @return WP_Error|true True when data has been saved correctly,
+ 	 * error otherwise.
+	*/
+	static function insert_widget_in_sidebar( $widget_id, $widget_options, $sidebar ) {
+		// Retrieve sidebars, widgets and their instances
+		$sidebars_widgets = get_option( 'sidebars_widgets', array() );
+		$widget_instances = get_option( 'widget_' . $widget_id, array() );
+
+		// Retrieve the key of the next widget instance
+		$numeric_keys = array_filter( array_keys( $widget_instances ), 'is_int' );
+		$next_key = $numeric_keys ? max( $numeric_keys ) + 1 : 2;
+
+		// Add this widget to the sidebar
+		if ( ! isset( $sidebars_widgets[ $sidebar ] ) ) {
+			$sidebars_widgets[ $sidebar ] = array();
+		}
+		$sidebars_widgets[ $sidebar ][] = $widget_id . '-' . $next_key;
+
+		// Add the new widget instance
+		$widget_instances[ $next_key ] = $widget_options;
+
+		// Store updated sidebars, widgets and their instances
+		if ( ! ( update_option( 'blogname', $data['siteTitle'] ) || get_option( 'blogname' ) == $data['siteTitle'] )
```

Comment:
This line seems off and unnecessary - we even don't have a `$data` here. This will probably throw an error when inserting a sidebar (although insertion will be successful).